GeneralsMD/Code/GameEngine/Include/GameNetwork/NetPacketStructs.h

```diff
@@ -0,0 +1,415 @@
+	UnsignedInt disconnectFrame;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Packet Router Command Packets
+////////////////////////////////////////////////////////////////////////////////
+
+// Packet router query command packet
+struct NetPacketRouterQueryCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+// Packet router ack command packet
+struct NetPacketRouterAckCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Base Packed Struct for All Command Messages
+// Contains the minimum common fields that all command messages share
+////////////////////////////////////////////////////////////////////////////////
+
+struct PackedNetCommandMsg {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Variable-Length Packet Headers
+// These structs represent the fixed portion of packets with variable data
+////////////////////////////////////////////////////////////////////////////////
+
+// Chat command header (variable: text follows)
+struct NetPacketChatCommandHeader {
+	NetPacketCommandTypeField commandType;
+	NetPacketFrameField frame;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketCommandIdField commandId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+struct NetPacketDisconnectChatCommandHeader {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Derived Packed Structs for getByteCount() calculations
+// These structs inherit from PackedNetCommandMsg and add command-specific fields
+////////////////////////////////////////////////////////////////////////////////
+
+struct PackedNetChatCommandMsg : public PackedNetCommandMsg {
+	NetPacketFrameField frame;
+	NetPacketCommandIdField commandId;
+};
+
+struct PackedNetDisconnectChatCommandMsg : public PackedNetCommandMsg {
+};
+
+// Game command packed struct (variable: game message data follows)
+struct PackedNetGameCommandMsg : public PackedNetCommandMsg {
+	NetPacketFrameField frame;
+	NetPacketCommandIdField commandId;
+};
+
+// File command packed struct (variable: filename and file data follow)
+struct PackedNetFileCommandMsg : public PackedNetCommandMsg {
+	NetPacketCommandIdField commandId;
+};
+
+// File announce command packed struct (variable: filename and metadata follow)
+struct PackedNetFileAnnounceCommandMsg : public PackedNetCommandMsg {
```

Comment:
Inconsistent naming. Some of the packed structures are prefixed with "Packed" and others are not.